app/domain/authentication/authn_jwt/signing_key/public_signing_keys.rb

```diff
@@ -0,0 +1,41 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # This class is a POJO class presents public-keys structure
+      class PublicSigningKeys
+        include ActiveModel::Validations
+        include AttrRequired
+
+        VALID_TYPES =  %w[jwks].freeze
+
+        attr_required(:type, :value)
+
+        validates(*required_attributes, presence: true)
+        validates(:type, inclusion: { in: VALID_TYPES, message: "'%{value}' is not a valid public-keys type" })
+        validate(:validate_value_is_jwks, if: -> { @type == "jwks" })
+
+        def initialize(hash)
+          raise Errors::Authentication::AuthnJwt::InvalidPublicKeys.new("the value is not in valid JSON format") unless
```

Comment:
Can "the value is not in valid JSON format" be moved to errors.rb

Reply:
Errors at the moment contains only exceptions
It's a variable part of general `InvalidPublicKeys` exception text...